File: 프로그래머스/LV2/코딩테스트 연습/연습문제/점 찍기/점 찍기.py
from math import sqrt

def solution(k, d):
    xs, ys = [], []

    for a in range(d//k+1):
        xs.append(a*k)
    
    for x in xs:
        ys.append(sqrt(d**2 - x**2) // k + 1)

    return sum(ys)

File: 프로그래머스/LV2/코딩테스트 연습/연습문제/점 찍기/점 찍기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <numeric>
using namespace std;

long long solution(int k, int d) {
    long long answer = 0;
    vector<int> xs,ys;

    for(int a = 0; a <= d/k; a++)xs.push_back(a*k);
    for(int x : xs)ys.push_back((int)(sqrt(pow(d,2) - pow(x,2)) / k) + 1);

    return accumulate(ys.begin(), ys.end(), 0LL);
}